ClientRuntimes/CSharp/Microsoft.Rest.ClientRuntime/BatchDelegatingHandler.cs

```diff
@@ -0,0 +1,185 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Rest
+{
+    /// <summary>
+    /// HTTP batch handler.
+    /// It expects the caller to dispose HttpRequestMessage and HttpResponseMessage.
+    /// It assumes the web service is supporting batch requests as per this MSDN article:
+    /// https://blogs.msdn.microsoft.com/webdev/2013/11/01/introducing-batch-support-in-web-api-and-web-api-odata/
+    /// i.e. requests are batched into a single request inside the multipart content.
+    /// </summary>
+    public class BatchDelegatingHandler : DelegatingHandler
+    {
+        /// <summary>
+        /// HTTP method (e.g. POST or PUT) to use in issuing a batch call.
+        /// </summary>
+        private readonly HttpMethod batchMethod;
+
+        /// <summary>
+        /// URL to issue the batch call to.
+        /// </summary>
+        private readonly Uri batchURL;
```

Comment:
What are batchUri and batchMethod actually used for?  Are these supposed to be used to filter the incoming requests?  I don't see this happening in the implementation, so curious what the purpose is.